src/optimisation/linear-problem-mpsolver-impl/include/antares/optimisation/linear-problem-mpsolver-impl/linearProblem.h

```diff
@@ -79,11 +79,12 @@ class OrtoolsLinearProblem: public LinearProblemApi::ILinearProblem
     bool isMaximization() const override;
 
     OrtoolsMipSolution* solve(bool verboseSolver) override;
-    void WriteLP(const std::string& filename) const override;
 
     double infinity() const override;
     bool isLP() const override;
 
+    friend void Write(const OrtoolsLinearProblem& problem, const std::filesystem::path& path);
```

Comment:
Here we still have dependency to an external **write** free function.
We should remove this friendship. 
This will cause a compilation error.
To make it go away,  we should make the following method **public** : 
```c++
operations_research::MPSolver* MpSolver() const;
```
and use it inside : 
```c++
void Write(const OrtoolsLinearProblem& problem, const std::filesystem::path& path)
```
If you hesitate to make **MpSolver()** public, don't : it **public** in class **LegacyOrtoolsLinearProblem**